### PR TITLE
[Heartbeat] Fix testing pipeline for Windows

### PR DESCRIPTION
**Pull request by emilioalvap**
Fix windows test pipeline by reflecting browser monitor build constraints in the include file

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

N/A

## How to test this PR locally

- Run `mage update` and check `x-pack/heartbeat/include/list.go` is constrained to `linux || darwin || synthetics` build tags.

## Related issues

Closes https://github.com/elastic/beats/issues/40502.

### CODE REVIEW

File: .buildkite/x-pack/pipeline.xpack.heartbeat.yml [hunk 1]
```diff
@@ -111,7 +111,7 @@ steps:
 
       - label: ":windows: x-pack/heartbeat: Win 2016 Unit Tests"
         key: "mandatory-win-2016-unit-tests"
-        skip: "skipping due to elastic/beats#23957 and elastic/beats#23958"
+        #skip: "skipping due to elastic/beats#23957 and elastic/beats#23958"
```
Comment (alexsapran) [inline .buildkite/x-pack/pipeline.xpack.heartbeat.yml:114]: Why are we commenting the `skip`?
If this PR fixes this then we should remove the skip, not?

Comment (emilioalvap) [reply .buildkite/x-pack/pipeline.xpack.heartbeat.yml]: @alexsapran I was just testing build flags, skip comment will be removed in the final commit

**Review by kaiyan-sheng — APPROVED** [review]

**Comment by emilioalvap** [comment]
@elastic/ingest-eng-prod could you provide a review? Thanks!

File: .buildkite/x-pack/pipeline.xpack.heartbeat.yml [hunk 2]
```diff
@@ -111,7 +111,7 @@ steps:
 
       - label: ":windows: x-pack/heartbeat: Win 2016 Unit Tests"
         key: "mandatory-win-2016-unit-tests"
-        skip: "skipping due to elastic/beats#23957 and elastic/beats#23958"
+        skip: "skipping due missing deps"
```
Comment (alexsapran) [inline .buildkite/x-pack/pipeline.xpack.heartbeat.yml:114]: Can we have a tracking issue for these missing dependencies to have a future reference?

Comment (emilioalvap) [inline .buildkite/x-pack/pipeline.xpack.heartbeat.yml:114]:
```suggestion
        skip: "skipping due missing deps, elastic/ingest-dev#3844"
```

**Comment by emilioalvap** [comment]
@elastic/ingest-eng-prod could you provide a review? Thanks!

**Review by rowlandgeoff — APPROVED** [review]
👍

**Pull request merged**